applications/matter_weather_station/src/app_task.cpp

```diff
@@ -384,7 +384,7 @@ CHIP_ERROR AppTask::Init()
 	k_timer_start(&sMeasurementsTimer, K_MSEC(kMeasurementsIntervalMs), K_MSEC(kMeasurementsIntervalMs));
 
 #ifdef CONFIG_MCUMGR_TRANSPORT_BT
-	Nrf::GetDFUOverSMP().StartServer();
+	PlatformMgr().ScheduleWork([](intptr_t) { Nrf::GetDFUOverSMP().StartServer(); });
```

Comment:
Why to do this in the Matter thread? If we need to do that after `StartServer` maybe let's move it after this call, e.g. to the `StartApp` after `Init`.

Reply:
I guess it's because the initialization of the SMP server takes place in the `Nrf::Matter::PrepareServer` that schedules all operation to the CHIP Thread. That being said, I'd suggest to place `Nrf::GetDFUOverSMP().StartServer()` in mPostServerInitClbk and feed it to the `Nrf::Matter::PrepareServer` call.

Reply:
```suggestion
		ReturnErrorOnFailure(Nrf::Matter::PrepareServer(Nrf::Matter::InitData{ .mPostServerInitClbk = [] {
#ifdef CONFIG_MCUMGR_TRANSPORT_BT
		Nrf::GetDFUOverSMP().StartServer();
#endif
		return CHIP_NO_ERROR;
	} }));
```

Reply:
Ok, I agree that we can do it this way, but I hope that we documented InitData properly because assigning the lambda function to a strange field of the InitData struct is crazy, especially for users who do not use C++.

Reply:
Each field of IniData is documented:
https://github.com/nrfconnect/sdk-nrf/blob/main/samples/matter/common/src/app/matter_init.h#L36
But we can always extend the docs if needed.